src/main/kotlin/com/openlattice/chronicle/services/timeusediary/TimeUseDiaryService.kt

```diff
@@ -0,0 +1,188 @@
+        submissionsIds: Set<UUID>
+    ) {
+        TODO("Not yet implemented")
+    }
+
+    /* -------- SQL helpers -------- */
+
+    private fun executeInsertTimeUseDiarySql(
+        connection: Connection,
+        tudId: UUID,
+        organizationId: UUID,
+        studyId: UUID,
+        participantId: String,
+        responses: List<TimeUseDiaryResponse>
+    ) {
+        val preparedStatement = connection.prepareStatement(insertTimeUseDiarySql)
+        var index = 1
+        preparedStatement.setObject(index++, tudId)
+        preparedStatement.setObject(index++, organizationId)
+        preparedStatement.setObject(index++, studyId)
+        preparedStatement.setString(index++, participantId)
+        preparedStatement.setObject(index, objectMapper.writeValueAsString(responses), Types.OTHER)
+        preparedStatement.execute()
+    }
+
+    private fun executeGetSubmissionByDateSql(
+        connection: Connection,
+        organizationId: UUID,
+        studyId: UUID,
+        participantId: String,
+        startDate: OffsetDateTime,
+        endDate: OffsetDateTime
+    ): ResultSet {
+        val preparedStatement = connection.prepareStatement(getSubmissionsByDateSql)
+        var index = 1
+        preparedStatement.setObject(index++, organizationId)
+        preparedStatement.setObject(index++, studyId)
+        preparedStatement.setString(index++, participantId)
+        preparedStatement.setObject(index++, startDate)
+        preparedStatement.setObject(index, endDate)
+        return preparedStatement.executeQuery()
+    }
+
+    /**
+     * SQL String to create a [java.sql.PreparedStatement] to submit a study response for a participant
+     *
+     * @param tudId             Identifies the Time Use Diary submission
+     * @param organizationId    Identifies the organization from which to retrieve submissions
+     * @param studyId           Identifies the study from which to retrieve submissions
+     * @param participantId     Identifies the participant for whom to retrieve submissions
+     * @param responses         List of survey responses for a Time Use Diary study
+     */
+    private val insertTimeUseDiarySql = """
+            INSERT INTO ${TIME_USE_DIARY_SUBMISSION.name} 
+            VALUES ( ?, ?, ?, ?, '${LocalDateTime.now()}', ? )
+            """.trimIndent()
+
+    /**
+     * SQL String to create a [java.sql.PreparedStatement] to retrieve study submissions for a participant within a date range
+     *
+     * @param organizationId    Identifies the organization from which to retrieve submissions
+     * @param studyId           Identifies the study from which to retrieve submissions
+     * @param participantId     Identifies the participant for whom to retrieve submissions
+     * @param startDate         Date that submissions must be submitted after or on
+     * @param endDate           Date that submissions must be submitted before or on
+     */
+    private val getSubmissionsByDateSql = """
```

Comment:
You can add an index to speed up this query, as long as you keep submission_date the last column in the index definition.

Reply:
But it's low pri as these queries will be super fast as is anyway.

Reply:
Working to finish time use diaries downloads currently, but will come back around to adding this index afterwards.